ts/features/payments/common/store/reducers/paymentsBackoffRetryReducer.ts

```diff
@@ -0,0 +1,46 @@
+import { getType } from "typesafe-actions";
+import { Action } from "../../../../../store/actions/types";
+import {
+  clearPaymentsBackoffRetry,
+  increasePaymentsBackoffRetry
+} from "../actions";
+import { PAYMENTS_BACKOFF_SECONDS_DELAYS } from "../../utils/backoffRetry";
+import { PaymentsBackoffRetryValue } from "../../utils/types";
+
+export type PaymentsBackoffRetryState = {
+  [key: string]: PaymentsBackoffRetryValue | undefined;
+};
+
+const INITIAL_STATE: PaymentsBackoffRetryState = {};
+
+const reducer = (
+  state: PaymentsBackoffRetryState = INITIAL_STATE,
+  action: Action
+): PaymentsBackoffRetryState => {
+  switch (action.type) {
+    case getType(increasePaymentsBackoffRetry):
+      const retryCount =
```

Comment:
To enhance the readability of this block, I recommend refactoring the code as follows:

```ts
      const currentRetryCount = state[action.payload]?.retryCount ?? 0;
      const retryCount = currentRetryCount + 1;
      const allowedRetryTimestamp =
        Date.now() + PAYMENTS_BACKOFF_SECONDS_DELAYS[retryCount] * 1000;

      return {
        ...state,
        [action.payload]: {
          retryCount,
          allowedRetryTimestamp
        }
      };
```

The  `1000` value could be stored in a constant called `SECONDS_TO_MILLISECONDS`